src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,288 @@
+
+        (self.__n_relevant,
+         self.__n_queries,
+         self.__index2imgid,
+         self.__relevant_arr,
+         self.__relevance_scores) = self.__parse_match_csv()
+
+        self._imgid2paths = dict(zip(self.__img_paths[self.__img_list_map_column['img_id']],
+                                     self.__img_paths[self.__img_list_map_column['image_path']]))
+
+        if len(self._imgid2paths) != len(self.__img_paths):
+            raise ValueError('Image csv have the same id for different image paths')
+
+        # filtering (save only img_id that in match.csv)
+        self._imgid2paths = {img_id: self._imgid2paths[img_id] for img_id in self.__index2imgid.values()}
+        self._data_len = self.__n_queries + self.__n_relevant
+
+        self.__use_gallery = use_gallery
+        self.__gallery_folder = gallery_folder
+        self.__gallery_list_csv_path = gallery_list_csv_path
+
+        if self.__use_gallery:
+            if self.__gallery_folder is None:
+                raise ValueError('Argument `gallery_folder` is None, please send path to gallery_folder')
+            if self.__gallery_list_csv_path is None:
+                raise ValueError('Argument `gallery_list_csv_path` is None, please send path to gallery_list_csv_path')
+
+            self.__gallery_paths = pd.read_csv(Path(self.__gallery_folder) / self.__gallery_list_csv_path,
+                                               usecols=[self.__gallery_map_column['gallery_id'], 
+                                                        self.__gallery_map_column['gallery_path']],
+                                               dtype={self.__gallery_map_column['gallery_id']: int, 
+                                                      self.__gallery_map_column['gallery_path']: str})
+            self.__gallery_imgid2paths = dict(zip(self.__gallery_paths[self.__gallery_map_column['gallery_id']], 
+                                                  self.__gallery_paths[self.__gallery_map_column['gallery_path']]))
+
+            if len(self.__gallery_imgid2paths) != len(self.__gallery_paths):
+                raise ValueError('Gallery csv have the same id for different image paths')
+
+            self.__n_gallery = 0
+            self._gallery_index2imgid = {}
+            for img_id in self.__gallery_imgid2paths:
+                self._gallery_index2imgid[self._data_len + self.__n_gallery] = img_id
+                self.__n_gallery += 1
+
+            self._data_len += self.__n_gallery
+
+        self.__imgid2index = {}
+        for index, img_id in self.__index2imgid.items():
+            if img_id not in self.__imgid2index:
+                self.__imgid2index[img_id] = index
+
+        self.__scores, self.__is_query = self.__get_targets()
+
+    def __getitem__(self, index: int) -> dict:
+        """Get item sample
+
+        Returns:
+            sample: dict, where
+            sample['image'] - image
```

Comment:
`Tensor, representing image after augmentations and transformations, dtype=image_dtype`